### PR TITLE
Fix for latest's Nim stricter effect tracking

### PR DESCRIPTION
**Pull request by planetis-m**
Ready

### CODE REVIEW

**Comment by nitely** [comment]
out of curiosity, what if `replace` did raise an error? is there a way to say "raises XYZ + `by` function exceptions? or more broadly my function effects + the passed function effects?

File: src/regex.nim [hunk 1]
```diff
@@ -828,12 +828,15 @@ func replace*(
...
+when not defined(nimHasEffectsOf):
+  {.pragma: effectsOf.}
+
 func replace*(
   s: string,
   pattern: Regex,
   by: proc (m: RegexMatch, s: string): string,
   limit = 0
-): string {.inline, raises: [].} =
+): string {.inline, effectsOf: by.} =
```
Comment (xflywind) [inline src/regex.nim:839]:
```suggestion
): string {.inline, raises: [], effectsOf: by.} =
```

Comment (nitely) [reply src/regex.nim]: this makes more sense

**Comment by nitely** [comment]
I'll merge it once tests pass (if I don't, please ping me)

**Comment by xflywind** [comment]
Thank you! I hope a tag will come soon.

**Comment by Araq** [comment]
> out of curiosity, what if replace did raise an error? is there a way to say "raises XYZ + by function exceptions? or more broadly my function effects + the passed function effects?

Then you leave out the `.raises:[]` part.

**Pull request merged**